frontend/app/professor/simulation-builder/page.tsx

```
@@ -869,13 +1345,111 @@ const handleSave = async (): Promise<number | null> => {
- const handlePlayScenario = async () => {
-   if (!autofillResult) {
-     alert("No scenario data to play. Please upload and process a PDF first.");
-     return;
+ // Auto-save to localStorage
+ const STORAGE_KEY = 'simulationBuilderDraft';
+ 
+ // Use refs to store latest values for save function
+ const formDataRef = useRef({
+   name,
+   description,
+   studentRole,
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)